### PR TITLE
53 improve roll

### LINKED ISSUE #53: More concise output for !r
So that the game master has a better overview in the chat, the following would be nice:

Input (from Peter): `!r 12+III Menschenkenntnis`
Output:
```
Peter Menschenkenntnis 30
+12+18(2@[20, 18, 12]) # this line greyed out or smaller if possible
```

If possible, it might be good to remove the input from the chat. This would keep it clearer. (However, it might be difficult to track errors in the input?) In any case, the extra message: "Rolling ..." gone, I think.

### PR DESCRIPTION
**Pull request by lukruh**
closes #53
depends on #77

most commits are from #77 so I'd wait until 77 is merged.

### CODE REVIEW

File: cogs/general/parse_die.py [hunk 1]
```diff
@@ -22,30 +21,41 @@ def roll_die(dice_number, sides, at=None):
 
 
 def parse_single_value(text):
-    if not "d" in text and not "w" in text:
+    if not "d" in text:
         return text, int(text)
     else:
         if "@" in text:
             at, dice_roll = text.split("@")
-            dice_number, sides = re.split("d|w", dice_roll.lower())
```
Comment (XaverStiensmeier) [inline cogs/general/parse_die.py:30]: why did you remove 'w'? Some German users might appreciate to be able to roll 2w6

File: cogs/general/parse_die.py [hunk 2]
```diff
@@ -22,30 +21,41 @@ def roll_die(dice_number, sides, at=None):
...
+            dice_number, sides = re.split("d", dice_roll.lower())
             throws, result = roll_die(dice_number=int(dice_number), sides=int(sides), at=int(at))
-            return f"{result}({at}@{throws})", result
+            throws_str = [f"__**{d}**__" if a==int(at)-1 else f"{d}" for a, d in enumerate(throws)]
+            at_str = f"[{', '.join(throws_str)}]"
+            # return f"{result}({at}@{throws})", result
+            return at_str, result
         else:
-            dice_number, sides = re.split("d|w", text.lower())
```
Comment (XaverStiensmeier) [inline cogs/general/parse_die.py:34]: Same question as above

File: cogs/general/parse_die.py [hunk 3]
```diff
@@ -22,30 +21,41 @@ def roll_die(dice_number, sides, at=None):
...
+            dice_number, sides = re.split("d", text.lower())
             throws, result = roll_die(dice_number=int(dice_number), sides=int(sides))
-            return f"{result}({throws})", result
+            # return f"{result}({throws})", result
+            return "[" + "+".join([f'**{d}**' for d in throws]) + "]", result
+
 
 def parse_roll(text):
+    text = text.lower().replace("w", "d")
```
Comment (XaverStiensmeier) [inline cogs/general/parse_die.py:43]: Ah I see. You just replaced 'w' with 'd'. Very neat :)

File: cogs/general/parse_die.py [hunk 4]
```diff
@@ -22,30 +21,41 @@ def roll_die(dice_number, sides, at=None):
...
     total_result = 0
     total_result_str = ""
     addends = text.split("+")
+    img = None
```
Comment (XaverStiensmeier) [inline cogs/general/parse_die.py:47]: I like the image addition.

File: cogs/generalCog.py [hunk 5]
```diff
@@ -11,16 +11,21 @@
 from cogs.general import parse_die
 
 cards = [os.path.splitext(filename)[0] for filename in os.listdir(RESOURCES/"manoeverkarten")]
-NAMED_ROLLS = [
+NAMED_ROLLS = [  # TODO: should this be part of settings?
```
Comment (XaverStiensmeier) [inline cogs/generalCog.py:14]: Probably.

**Review by XaverStiensmeier — APPROVED** [review]
I haven't really tested the functionality, but just read the code. I guess a real functionallity test is most worth it when we have merged everything into dev to check if something broke down.

**Comment by lukruh** [comment]
> I haven't really tested the functionality, but just read the code. I guess a real functionallity test is most worth it when we have merged everything into dev to check if something broke down.

Yes I added v0.2 and v0.3 mile stones, thats where I'd test everything together in the dev branch before it goes to the main. So everything that need to be considered later should be somewhere assigned to the milestone for final review.

File: cogs/general/parse_die.py (hunk 4 above)
Comment (lukruh) [reply cogs/general/parse_die.py:47]: I hosted the images on a webserver would probably good to add them to the repo as well.
